test-data/unit/check-classvar.test

```diff
@@ -0,0 +1,44 @@
+[case testAssignmentOnClass]
+from typing import ClassVar
+class A:
+    x = 1  # type: ClassVar[int]
+A.x = 2
+
+[case testAssignmentOnInstance]
+from typing import ClassVar
+class A:
+    x = 1  # type: ClassVar[int]
+A().x = 2
+[out]
+main:4: error: Illegal assignment to class variable
+
+[case testAssignmentOnSubclass]
+from typing import ClassVar
+class A:
+    x = 1  # type: ClassVar[int]
+class B(A):
+    pass
+B().x = 2
+[out]
+main:6: error: Illegal assignment to class variable
+
+[case testReadingFromInstance]
+from typing import ClassVar
+class A:
+    x = 1  # type: ClassVar[int]
+A().x
+
+[case testTypecheck]
+from typing import ClassVar
+class A:
+    x = 1  # type: ClassVar[int]
+y = A.x  # type: int
+
+[case testInfer]
+from typing import ClassVar
+class A:
+    x = 1  # type: ClassVar[int]
+y = A.x
+reveal_type(y)
+[out]
+main:5: error: Revealed type is 'builtins.int'
```

Comment:
I think more tests are needed here, some ideas:
* More tests on accessing class vars  (through instances, subclasses, and instances of subclasses)
* Class vars with complex types: user defined types, generic collections
* Test behaviour of class vars with type variables in user defined generic classes
* Check correct access and inference for the above point
* Check more overriding options: overriding by an instance variable in subclass, and by assigning to an instance of subclass (I believe both should be errors)
* Test behaviour on attempts to define and/or override class vars inside method bodies.
* Maybe multiple inheritance
* Add also few tests that combine more than one of the above ideas